include/sparrow/struct_array.hpp

```diff
@@ -508,6 +558,24 @@ namespace sparrow
         );
         return arrow_proxy{std::move(arr), std::move(schema)};
     }
+
+    template <layout_or_array A>
+    void struct_array::add_child(A&& child)
+    {
+        SPARROW_ASSERT_TRUE(child.size() == size());
+        auto [array, schema] = extract_arrow_structures(std::forward<A>(child));
+        get_arrow_proxy().add_child(std::move(array), std::move(schema));
+        m_children = make_children();
+    }
+
+    template <layout_or_array A>
+    void struct_array::set_child(A&& child, size_t index)
+    {
+        SPARROW_ASSERT_TRUE(child.size() == size());
+        auto [array, schema] = extract_arrow_structures(std::forward<A>(child));
+        get_arrow_proxy().set_child(index, std::move(array), std::move(schema));
+        m_children = make_children();
```

Comment:
Can we avoid rebuilding the whole children list, and just rebuild the one we changed?